app/views/courses/EnrollmentsView.coffee

```diff
@@ -53,28 +55,54 @@ module.exports = class EnrollmentsView extends RootView
 
   onceClassroomsSync: ->
     for classroom in @classrooms.models
-      @members.fetch({
-        remove: false
-        url: "/db/classroom/#{classroom.id}/members"
-      })
+      @supermodel.trackRequests @members.fetchForClassroom(classroom, {remove: false, removeDeleted: true})
 
-  membersSync: ->
+  calculateEnrollmentStats: ->
+    @removeDeletedStudents()
     @memberEnrolledMap = {}
     for user in @members.models
       @memberEnrolledMap[user.id] = user.get('coursePrepaidID')?
-    @classroomNotEnrolledMap = {}
-    @totalNotEnrolled = 0
+      
+    @totalEnrolled = _.reduce @members.models, ((sum, user) ->
+      sum + (if user.get('coursePrepaidID') then 1 else 0)
+    ), 0
+    
+    @totalNotEnrolled = _.reduce @members.models, ((sum, user) ->
+      sum + (if not user.get('coursePrepaidID') then 1 else 0)
+    ), 0
+    
+    @classroomEnrolledMap = _.reduce @classrooms.models, ((map, classroom) =>
+      enrolled = _.reduce classroom.get('members'), ((sum, userID) =>
+        sum + (if @members.get(userID).get('coursePrepaidID') then 1 else 0)
+      ), 0
+      map[classroom.id] = enrolled
+      map
+    ), {}
+    
+    @classroomNotEnrolledMap = _.reduce @classrooms.models, ((map, classroom) =>
+      enrolled = _.reduce classroom.get('members'), ((sum, userID) =>
+        sum + (if not @members.get(userID).get('coursePrepaidID') then 1 else 0)
+      ), 0
+      map[classroom.id] = enrolled
+      map
+    ), {}
+    
+    true
+    
+  removeDeletedStudents: (e) ->
     for classroom in @classrooms.models
-      @classroomNotEnrolledMap[classroom.id] = 0
-      for memberID in classroom.get('members')
-        @classroomNotEnrolledMap[classroom.id]++ unless @memberEnrolledMap[memberID]
-      @totalNotEnrolled += @classroomNotEnrolledMap[classroom.id]
-    @numberOfStudents = @totalNotEnrolled
-    @render?()
+      _.remove(classroom.get('members'), (memberID) =>
+        not @members.get(memberID) or @members.get(memberID)?.get('deleted')
+      )
+    true
```

Comment:
What's this for? TeacherClassView has something similar at the end of its `removeDeletedStudents`.